worlds/sc2/Regions.py

```diff
@@ -26,7 +27,7 @@ def create_regions(
     * int The number of missions in the world
     * str The name of the goal location
     """
-    mission_order_type: int = get_option_value(world, "mission_order")
+    mission_order_type: MissionOrder = get_option_value(world, "mission_order")
```

Comment:
mypy complains around these type checks, recommend using the modern-style `world.options.mission_order.value` style. Technically, the value is an `int`, not a `MissionOrder`. Otherwise, mypy sees that `get_option_value()` returns `int | FrozenSet` and reports a problem.

Reply:
I don't _want_ to make it an int, though, I want it to be clear that it's a MissionOrder.  `world.options.mission_order` seems fine.

Reply:
The difference is that `get_option_value(world, "mission_order")` supports cases where `world` is `None`. (IIRC it must be used in Locations and logic)

Reply:
It shouldn't be a problem in Regions?

Reply:
My stance is that if the method supports `None`, that should be reflected in the typehint with `Optional['SC2World']`. I don't think this method needs to support `None`, and currently it doesn't. In general, I'd still prefer just an explicit null-check followed by the `world.options` style because the type on `get_option_value` is `int|FrozenSet` which is more often than not wrong.

Reply:
I'm fine to call this resolved, I just don't have such a button in my UI (I think because I'm not listed as a maintainer / member / whatever in Ziktofel's repo?)